addons/config/hack/test.sh

```diff
@@ -43,6 +58,10 @@ function verifyAllAddonConfigTemplates() {
       done
 		done
 	done
+
+  # additionally, add test coverage for VSphereCPIConfig when iaas is tkgs, i.e. paravirtual mode
+	verifyAddonConfigTemplateForGVRWithIaas "cpi.tanzu.vmware.com" "v1alpha1" "vspherecpiconfig" "tkgs"
```

Comment:
**thought**: Instead of adding testcase into test.sh could we make this a bit more generic ?

i.e. Add addons/config/testcases/cpi.tanzu.vmware.com/v1alpha1/vsphere.yaml

---
iaas: vsphere

Add addons/config/testcases/cpi.tanzu.vmware.com/v1alpha1/tkgs.yaml

---
iaas: tkgs

And use this in testcase:

for each testcase in testcases:
TEST_RESULT=$(${TF_TOOL_DIR}/ytt --ignore-unknown-comments -f templates/${1}/${2}/${3}.yaml -v TKR_VERSION=${TKR_INPUT} -v GLOBAL_NAMESPACE=${NAMESPACE_INPUT} --data-value-file=<testcase-file>

Reply:
I think that's a good idea. Will add that. thanks!

Reply:
I've added a `config/testcases` folder that stores data values for the template, to make the test more generic. Could you review the change at your convenience? Thanks @shyaamsn